File: SOC/src/cc/game/soc/core/BotNodeTile.java
package cc.game.soc.core;

import cc.lib.game.IVector2D;

public class BotNodeTile extends BotNode {

	private Tile tile;
	private int index;
	
	public BotNodeTile() {}
	
	BotNodeTile(Tile tile, int index) {
		this.tile = tile;
		this.index = index;
	}

	@Override
	public Object getData() {
		return index;
	}

	@Override
	public String getDescription() {
		return "T(" + index + ") " + tile.toString();
	}
	
	public IVector2D getBoardPosition(Board b) {
    	return tile;
    }

	
}

File: SOC/src/cc/game/soc/core/BotNodeTile.kt
package cc.game.soc.core

import cc.lib.game.IVector2D

class BotNodeTile(val tile: Tile, index: Int) : BotNode() {

	override val data = index

	override val description: String= "T($index) $tile"

	override fun getBoardPosition(b: Board): IVector2D {
		return tile
	}
}